### ISSUE TITLE
Shell: parsing quotes inside comment inside command substituion

### ISSUE THREAD

**Issue by Herst** [google/code-prettify#453]
Examples with the bug:

```sh
foobar --ok `# '` --bug
```

```sh
foobar --ok `# "` --bug
```

```sh
foobar --ok "`# '`" --bug
```

```sh
foobar --ok $(# ') --bug
```

Real-life case, the bug report at: https://github.com/koalaman/shellcheck/issues/729 (not the bug reported there itself)

**Comment by amroamroamro** [comment]
here are the examples rendered live, with the bug in the last one: https://jsfiddle.net/amroamroamro/xkjhf5q9/

**Comment by Herst** [comment]
Oh, I'm, terribly sorry, twice actually, once for assuming that github uses `google-code-prettify `(somehow I got it mixed up with what the `shellcheck` web UI uses) and the secondly for giving you the wrong example. Apparently this is one of those rare cases where `$()` indeed behave differently from `` and the syntax highlighting *is* actually spot on when it comes to the last example! I'm terribly sorry for anyone's time I wasted.
